### PR TITLE
Adding medium connector

### PR DESCRIPTION
**Pull request by SanduDS**
Fixes https://github.com/wso2-enterprise/choreo/issues/5955

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi/medium/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+org = "ballerinax"
+name = "medium"
+version = "0.1.0"
+keywords = ["Medium", "Publication"]
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+authors = ["Ballerina"]
+license = ["Apache-2.0"]
+
+[build-options]
+observabilityIncluded = true
```
Comment (lnash94) [inline openapi/medium/Ballerina.toml:11]: Shall we add a new line here, if possible? Please check other places as well. :)

Comment (SanduDS) [reply openapi/medium/Ballerina.toml]: Added new line there. Thank you

File: openapi/medium/client.bal [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// under the License.
+
+import  ballerina/http;
+
+public type ClientConfig record {
+    http:BearerTokenConfig|http:OAuth2RefreshTokenGrantConfig authConfig;
+    http:ClientSecureSocket secureSocketConfig?;
+};
+
+# Medium’s unofficial API documentation using OpenAPI specification.
```
Comment (aneeshafedo) [inline openapi/medium/client.bal:24]: Please update this doc comments to a connector related description. Eg: `Client endpoint for Medium API`

Comment (SanduDS) [reply openapi/medium/client.bal]: Updated

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**